Controllers/Ingredient_ComponentsController.cs

```diff
@@ -48,45 +49,37 @@ public ActionResult Index(int IngredID)
             return View(Ingred_nut.ToList());
         }
         [HttpGet]
-        public JsonResult GetData(int ingredId)
+        public async Task<JsonResult> GetData(int ingredId)
         {
             int CustomerId = GetUserCustomerId(HttpContext.Session.GetString("CurrentUserCustomerId"));
             //BUILD VIEW FOR ANGULARJS RENDERING
-            var query =
-               from t1 in _recitopiaDbContext.Ingredient_Components.AsQueryable()
-               join t2 in _recitopiaDbContext.Ingredient.AsQueryable() on t1.Ingred_Id equals t2.Ingredient_Id into t2g
-               from t2 in t2g.DefaultIfEmpty() //DefaultIfEmpty is used for left joins
-               join t3 in _recitopiaDbContext.Components.AsQueryable() on t1.Comp_Id equals t3.Comp_Id
-               where t1.Ingred_Id == ingredId
-               where t1.Customer_Id == CustomerId
-               orderby t3.Component_Name
-               select new View_Angular_Ingredient_Components_Details()
-               {
-                   Id = t1.Id,
-                   Customer_Id = t1.Customer_Id,
-                   Ingred_Id = t1.Ingred_Id,
-                   Ingred_name = t2.Ingred_name,
-                   Comp_Id = t1.Comp_Id,
-                   Component_Name = t3.Component_Name,                  
-
-               };
-
-            List<View_Angular_Ingredient_Components_Details> ingredientcomponent = query.ToList();
-
-            if (ingredientcomponent != null)
-            {
-                return Json(ingredientcomponent);
-            }
-            return Json(new { Status = "Failure" });
+            var ingredientComponents = await _recitopiaDbContext.Ingredient_Components
+                .Include(ri => ri.Ingredients)
+                .Include(ri => ri.Components)
+                .Where(ri => ri.Customer_Id == CustomerId && ri.Ingred_Id == ingredId)
+                .Select(ri => new View_Angular_Ingredient_Components_Details()
+                {
+                    Id = ri.Id,
+                    Customer_Id = ri.Customer_Id,
+                    Ingred_Id = ri.Ingred_Id,
+                    Ingred_name = ri.Ingredients.Ingred_name,
+                    Comp_Id = ri.Comp_Id,
+                    Component_Name = ri.Components.Component_Name
+                })
+                .ToListAsync();
+
+            return ingredientComponents != null
+                ? Json(ingredientComponents)
+                : Json(new { Status = "Failure" });
         }
         // GET: Ingredient_Components/Details/5
-        public ActionResult Details(int? id)
+        public async Task<ActionResult> Details(int? id)
```

Comment:
IActionResult

Reply:
Should all ActionResult be changed to IActionResult?